content/docs/learn/typed-errors/working-with-typed-errors.md

```diff
@@ -1,47 +1,102 @@
-- [_Exception handling in Kotlin with Arrow_](https://www.youtube.com/watch?v=ipF540mBG9w) by Ramandeep Kaur
-- [_Por qué no uso excepciones en mi código_](https://www.youtube.com/watch?v=8WdprhzmQe4) by Raúl Raja and [Codely](https://codely.com/)
+We use the term _logical failure_ to describe a situation not deemed as successful in your domain, but that it's still within the realms of that domain.
+For example, if you are implementing a repository for users, not finding a user for a certain query is a logical failure.
+
+In contrast to logical failures we have _real exceptions_, which are problems, usually technical, which do not fit in the domain.
+For example, if the connection to the database suddenly drops, or the connection credentials are wrong.
+
+:::
+
+:::note Success and failure
+
+When talking about error handling, we often distinguish between _success_ or _happy path_, and _failure_.
+The former represents the case in which everything works as intended, whereas the latter represents a problem.
+Depending on the approach, the signature of the function only signals that a failure is possible,
+or additionally describes the range of problems that may arise.
 
 :::
 
-## Different types
+There are two main approaches to representing types in the signature of a function.
+Fortunately, Arrow provides a _uniform_ API to working with all of them, which is described in the rest of this section.
+
+The first approach is using a _wrapper type_, in which the return type of your function
+is nested within a larger type that provides the choice of error. 
+In that way the error is represented as a _value_.
+For example, the following signature expresses that the outcome of `findUser` is 
+of type `User` when successful, or `UserNotFound` when a logical failure is raised.
 
-There are three ways of working with errors (in addition to [_nullability_ and `Option`](../nullable-and-option), which model simple absence of value):
+```
+fun findUser(id: UserId): Either<UserNotFound, User>
+```
 
-- `Either<E, A>` represents a _computed value_ of _either_ a _logical failure_ of `E` or a _success_ value `A`.
+The Kotlin standard library includes a few wrapper types, but they are all restricted in the information they may include.
+Arrow introduces `Either` and `Ior`, both giving the developer the choice of type of logical failures, and reflecting that choice
+as their first type parameter.
 
-- `Ior<E, A>` represents a computed value of _either_ a _logical failure_ of `E` or a _success_ value `A` or **both** a success value of `A` together with a _logical failure_ of `E`.
+| Type | Failure | Simultaneous <br /> success and failure? | Kotlin stdlib. or Arrow? |
+|---|---------|------|---|
+| `A?` | No information | | <img src="https://upload.wikimedia.org/wikipedia/commons/3/37/Kotlin_Icon_2021.svg" style={{height: '20px'}} /> |
```

Comment:
@serras I wonder if we should be more specific using a subject, something like "value is null" for nullable, and "value is empty (`None`)" for `Option<A>`, "value is left with a generic type `E`"... etc.  I tried rendering the page and saw "No information", I thought there was a javascript problem 😅

Reply:
Good idea, I'll add this in the upcoming commit.